scripts/setup/get-icons.js

```diff
@@ -0,0 +1,175 @@
+'use strict';
+const { exec, execSync } = require('child_process');
+const fs = require('fs');
+const chalk = require('chalk');
+const path = require('path');
+
+const fileExists = fs.existsSync('./.resources.json');
+if (!fileExists) {
+  console.log('File ./.resources.json not exists');
+  return;
+}
+
+const resourcesData = JSON.parse(fs.readFileSync('./.resources.json'));
+const iconFolder = 'public/assets/icons/';
+
+if (!resourcesData.hasOwnProperty('repoUrl') && !resourcesData.hasOwnProperty('path')) {
+  return;
+}
+
+class IconManagerBase {
+  constructor(resourcesData, iconFolder, tempFolder) {
+    this.resourcesData = resourcesData;
+    this.iconFolder = iconFolder;
+    this.tempFolder = tempFolder;
+    this.rootPath = '../../';
+  }
+}
+
+class ISparseCheckoutGitV225 extends IconManagerBase {
+  constructor(resourcesData, iconFolder, tempFolder) {
+    super(resourcesData, iconFolder, tempFolder);
+  }
+
+  execute = () => {
+    execSync(
+      `git clone "${this.resourcesData.repoUrl}" --no-checkout  --depth=1 --single-branch --branch ${this.resourcesData.branch} ${this.tempFolder} -o repo`,
+      {
+        stdio: [0, 1, 2],
+        cwd: path.resolve(__dirname, this.rootPath),
+      },
+    );
+
+    execSync(
+      `git sparse-checkout init --cone &&
+            git config core.sparseCheckout 1 &&
+            git sparse-checkout set ${this.resourcesData.folder} &&
+            git checkout`,
+      {
+        stdio: [0, 1, 2],
+        cwd: path.resolve(__dirname, `../../${this.tempFolder}`),
+      },
+    );
+  };
+}
+
+class ISparseCheckoutGitV217 extends IconManagerBase {
```

Comment:
Although this is not typescript, but let's still try to follow the style guide https://basarat.gitbook.io/typescript/styleguide#interface

Reply:
Got it! Thanks for the feedback!